tests/integration/snmp-hide-password-test.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+# This test is expected to be run as root in a container(docker, podman, toolbox, apptainer, (s)chroot):
+# It is currently started from .github/workflows/alpine-python2.yml
+# Precondition: python with all dependencies from requirements.txt is installed
+# WARNING: This is running as user root in the container it runs in!
+#
+# A quick way to run it is:
+# act -W .github/workflows/alpine-python2.yml
+#
+# exit on any error
+set -o errexit
+set -o pipefail
+if [[ -n "$TRACE" ]]; then set -o xtrace; fi
+set -o nounset
+: ${PYTHON:=python2}
+
+# Prepare test container: Mock the files used by this test
+cp -a tests/integration/dom0-template/* /
+
+# Enter a clean test environment
+SRCDIR=$PWD
+rm -rf   .tmp/tests/snmp
+mkdir -p .tmp/tests/snmp
+cd       .tmp/tests/snmp
+export PYTHONPATH=~-/tests/mocks
+
+# Check that mocking xen.lowlevel.xc works for this test
+$PYTHON -c "from xen.lowlevel.xc import Error, xc;xc().domain_getinfo()"
+
+# Run xen-bugtool --entries=xenserver-config to capture the SNMP files
+export XENRT_BUGTOOL_BASENAME=zip
+$PYTHON ~-/xen-bugtool -y --entries=xenserver-config --output=zip
+unzip -o -d. /var/opt/xen/bug-report/zip.zip
+
+cd $XENRT_BUGTOOL_BASENAME
+
+# Check filename entries in inventory.xml (in case a tool expects them):
+grep -q \
+    -e "filename=\"$XENRT_BUGTOOL_BASENAME/snmp_xs_conf.out\"" \
+    -e "filename=\"$XENRT_BUGTOOL_BASENAME/snmpd_xs_conf.out\"" \
+    -e "filename=\"$XENRT_BUGTOOL_BASENAME/snmpd_conf.out\"" \
+    -e "filename=\"$XENRT_BUGTOOL_BASENAME/etc/sysconfig/snmpd\"" \
+    inventory.xml
+
+# Check that snmp_xs_conf.out should not contain community or keys
+snmp_xs_conf_filter_str=$(grep 'snmpv2cpublic\|0xsnmpv3authenticationkey\|0xsnmpv3privacykey' /code/status-report/.tmp/tests/snmp/tar/snmp_xs_conf.out || true)
```

Comment:
The presence of strings can be checked using grep and its exit code. The test `"$snmp_xs_conf_filter_str" != ""` seems more complicated to me.

Reply:
Updated.